sys/uartpb/UartProtobuf.cpp

```diff
@@ -53,49 +54,83 @@ void UartProtobuf::uart_rx_cb(uint8_t data)
 
 void UartProtobuf::message_reader()
 {
+    uuid_t uuid;
     msg_t msg;
     msg_t msg_queue[8];
     msg_init_queue(msg_queue, 8);
 
     while (1) {
+        // Wait for a new message
         msg_receive(&msg);
+
+        // Read and check message length
         uint32_t message_length = (uint32_t)msg.content.value;
         assert(message_length <= UARTPB_INPUT_MESSAGE_LENGTH_MAX);
-        read_buffer_.clear();
-        ringbuffer_get(&rx_buf_, (char *)read_buffer_.get_base64_data(), message_length);
-        size_t res = read_buffer_.base64_decode();
-        if (res > 0 && message_handler_) {
-            message_handler_(read_buffer_);
+
+        // Read uuid
+        ringbuffer_get(&rx_buf_, (char *)&uuid, sizeof(uuid_t));
+        message_length -= sizeof(uuid_t);
+
+        // Check a handler corresponding to the uuid is registered
+        if (message_handlers_.count(uuid) != 1) {
+            std::cout << "Unknown message uuid: " << uuid << std::endl;
```

Comment:
Why not using COGIP_DEBUG_COUT ? (same in all the file)

Reply:
Because COGIP_DEBUG_COUT is for debug messages and this is an error/log message.

Reply:
Use COGIP_DEBUG_CERR :wink:

Reply:
Nope, still a debug macro...

Reply:
It was a bad idea to add these macros, we should have just wait to add the log module.

Reply:
Just create few macros not for debug like COGIP_COUT/COGIP_CERR.
It is completly acceptable to me to have this temporary macros.

Reply:
Or we can simply assume you let it as is until the log module is ready. What do you think ?

Reply:
Yes I prefer to wait for the log module.

Reply:
https://github.com/cogip/mcu-firmware/issues/78